tests/runtime/device_runtime_tests.cpp

```diff
@@ -113,19 +109,22 @@ void DeviceRuntimeTests::ProcessISB(const p_data_hdr_t &dataHdr, const uint8_t *
     }
 }
 
-void DeviceRuntimeTests::TestIsbGps(const p_data_hdr_t &dataHdr, const uint8_t *dataBuf)
+std::deque<DeviceRuntimeTests::msg_history_t>& DeviceRuntimeTests::AddMsgHistory(std::deque<DeviceRuntimeTests::msg_history_t> &hist, DeviceRuntimeTests::msg_history_t msgHist)
 {
-    msg_history_t &hist = m_hist.isb.gps1Pos;
-
-    copyDataPToStructP2(&m_gps1Pos, &dataHdr, dataBuf, sizeof(gps_pos_t));
+    hist.push_front(msgHist);
+    if (hist.size() > 3) { hist.pop_back(); }    // Keep 2, newest at front
+    return hist;
+}
 
-    // printf("ISB GpsPos1 (%d ms, %d wkday): %.*s", m_gps1Pos.timeOfWeekMs, utcWeekday, msgSize, msg);
+void DeviceRuntimeTests::TestIsbGps(const p_data_hdr_t &dataHdr, const uint8_t *dataBuf)
```

Comment:
Even though these are internal functions, having doxygen or at least high-level comments that describe the purpose of the test really make it easier for other to follow in your footsteps.  Its not really clear what these next few tests are doing, and why they are important.